turbine/src/broadcast_stage/broadcast_utils.rs

```diff
@@ -33,7 +48,23 @@ fn keep_coalescing_entries(
     // This slot is not over.
     last_tick_height < max_tick_height &&
     // We have not exceeded max batch byte count.
-    serialized_batch_byte_count < max_batch_byte_count
+    serialized_batch_byte_count < max_batch_byte_count &&
+    {
+        let bytes_to_fill_erasure_batch = data_shred_bytes_per_batch() - (serialized_batch_byte_count % data_shred_bytes_per_batch());
+        // We haven't tightly packed this batch.
+        (bytes_to_fill_erasure_batch as f64) > target_batch_pad_bytes()
+    }
+}
+
+// Dynamically determine the coalesce time based on the amount of entry data
+// that has been built up. If we only have a small amount, we can afford to wait
+// longer and avoid unnecessary padding. If we have a lot, we shouldn't wait as
+// long so we avoid delaying downstream validator replay.
+fn max_coalesce_time(serialized_batch_byte_count: u64, max_batch_byte_count: u64) -> Duration {
```

Comment:
maybe use f32 here? Not critical but there is totally no need for f64 in such code.

Reply:
done